src/compiler/compile/render_dom/index.ts

```diff
@@ -531,7 +531,10 @@ export default function dom(
 				constructor(options) {
 					super();
 
-					${css.code && b`this.shadowRoot.innerHTML = \`<style>${css.code.replace(regex_backslashes, '\\\\')}${css_sourcemap_enabled && options.dev ? `\n/*# sourceMappingURL=${css.map.toUrl()} */` : ''}</style>\`;`}
+					${css.code && b`
+						const style = document.createElement('style');
+						style.textContent = \`${css.code.replace(regex_backslashes, '\\\\')}${css_sourcemap_enabled && options.dev ? `\n/*# sourceMappingURL=${css.map.toUrl()} */` : ''}\`
+						this.shadowRoot.appendChild(style)`}
```

Comment:
If you want to avoid the `style` var, as this is compiled code, you can do:

```suggestion
						this.shadowRoot.appendChild(document.createElement('style')).textContent = \`${css.code.replace(regex_backslashes, '\\\\')}${css_sourcemap_enabled && options.dev ? `\n/*# sourceMappingURL=${css.map.toUrl()} */` : ''}\`
						`}
```

Reply:
Do you think it's worth compressing these into a single line? Presumably, if this goes through a bundler it will be more than capable of minifying the resultant code better than I can here, and for the purposes of development, I think it's a bit more readable having it on two separate lines.

Happy to update it if you think it's worth it though.